lib/collections.go

```diff
@@ -948,6 +1020,44 @@ func zipLists(arg0, arg1 ref.Val) ref.Val {
 	return types.NewRefValMap(types.DefaultTypeAdapter, m)
 }
 
+func mapKeys(val ref.Val) ref.Val {
+	mapK, ok := val.(traits.Mapper)
+	if !ok {
+		return types.ValOrErr(mapK, "no such overload")
+	}
+	n, _ := mapK.Size().(types.Int)
+	new := make([]ref.Val, 0, n)
+	if mapK.Size() != types.IntZero {
+		m, err := mapK.ConvertToNative(refValMap)
+		if err != nil {
+			return types.NewErr("unable to convert map to native: %v", err)
+		}
+		for key, _ := range m.(map[ref.Val]ref.Val) {
+			new = append(new, key)
+		}
```

Comment:
```suggestion
		canSort := true
		for k := range m.(map[ref.Val]ref.Val) {
			keys = append(keys, k)
			_, ok := k.(traits.Comparer)
			if !ok {
				canSort = false
			}
		}
		if canSort {
			sort.Slice(keys, func(i, j int) bool {
				return keys[i].(traits.Comparer).Compare(keys[j]) == types.Int(-1)
			})
		}	
```
This is not completely safe, if someone decides to make a `<map<dyn,dyn>>` it could fail, but I think that is a reasonable thing to do if they choose to do something so foolish.

Though note that this should make use of the iterator rather than a Go range.

Reply:
Updated as suggested with iterator